protocols/1520-gencell-pharma-part3/cleanup_PCR1.ot2.py

```diff
@@ -0,0 +1,144 @@
+    etanol = [well for row in tubes.rows()
+              for well in row][2:int(2+number_of_samples/2)]
+    beads = tubes.wells('A1')
+    buffer = tubes.wells('A2')
+
+    # mix and distribute magnetic beads to samples
+    m300.pick_up_tip(tips[tip_counter])
+    m300.mix(10, 300, beads)
+    m300.drop_tip()
+    tip_counter += 1
+    for s in samples:
+        m300.pick_up_tip(tips[tip_counter])
+        m300.transfer(90, beads, s, disposal_vol=5, new_tip='never')
+        m300.mix(10, 120, s)
+        m300.drop_tip()
+        tip_counter += 1
+
+    # incubate at room temperature for 10 minutes
+    m300.delay(minutes=10)
+
+    robot.pause("Place the sample rack on the magnetic module before "
+                "resuming. The magdeck will engage for 2 minutes before the "
+                "protocol resumes.")
+    magdeck.engage()
+    m300.delay(minutes=2)
+
+    # transfer out supernatant from samples
+    for i, s in enumerate(mag_samples):
+        m300.pick_up_tip(tips[tip_counter])
+        m300.transfer(140, s.bottom(2), waste_rack.wells(i), new_tip='never')
+        m300.drop_tip()
+        tip_counter += 1
+
+    # distribute ethanol to samples
+    m300.pick_up_tip(tips[tip_counter])
+    m300.distribute(200, etanol[0], mag_samples, new_tip='never')
+    m300.drop_tip()
+    tip_counter += 1
+
+    # incubate at room temperature for 30 seconds
+    m300.delay(seconds=30)
+
+    # transfer out supernatant from samples
+    for i, s in enumerate(mag_samples):
+        m300.pick_up_tip(tips[tip_counter])
+        m300.transfer(200,
+                      s.bottom(2),
+                      waste_rack.wells(i+number_of_samples),
+                      new_tip='never')
+        m300.drop_tip()
+        tip_counter += 1
+
+    # distribute ethanol to samples
+    m300.pick_up_tip(tips[tip_counter])
+    m300.distribute(200, etanol[1], mag_samples, new_tip='never')
+    m300.drop_tip()
+    tip_counter += 1
+
+    # incubate at room temperature for 30 seconds
+    m300.delay(seconds=30)
+
+    # transfer out supernatant from samples
+    for i, s in enumerate(mag_samples):
+        m300.pick_up_tip(tips[tip_counter])
+        m300.transfer(200,
+                      s.bottom(2),
+                      waste_rack.wells(i+number_of_samples*2),
+                      new_tip='never')
+        m300.drop_tip()
+        tip_counter += 1
+
+    # incubate at room temperature for 10 minutes
+    m300.delay(minutes=10)
+
+    magdeck.disengage()
+    robot.pause("Remove the sample rack from the magnetic module before "
+                "resuming.")
+
+    # distribute resuspension buffer to samples and mix
+    for s in samples:
+        m50.pick_up_tip(tips[tip_counter])
+        m50.transfer(27.5, buffer, s, disposal_vol=5, new_tip='never')
+        m50.mix(15, 20, s)
+        m50.drop_tip()
+        tip_counter += 1
+
+    # incubate at room temperature for 2 minutes
+    m300.delay(minutes=2)
```

Comment:
Not sure if this would work. The robot might skip the 2 min delay and go straight to pause. Maybe we should just ask the user to time it during the pause